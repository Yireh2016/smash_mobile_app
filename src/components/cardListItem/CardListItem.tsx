import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import { ChildrenProp } from '../../types/common/commonTypes';
import theme from '../../theme/theme';

interface CardListProps extends ChildrenProp {
  isLight?: boolean;
}
const CardListItem: React.FunctionComponent<CardListProps> = ({
  children,
  isLight = false,
}) => {
  const styles = StyleSheet.create({
    cardListContainer: {
      backgroundColor: isLight
        ? theme.palette.tertiary
        : theme.palette.primaryLight,
      borderRadius: 12,
      width: '100%',
    },
  });
  return <View style={styles.cardListContainer}>{children}</View>;
};

export default CardListItem;
