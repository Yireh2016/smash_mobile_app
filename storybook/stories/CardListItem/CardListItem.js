/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import CardListItem from '../../../src/components/cardListItem/CardListItem';
import TableRow from '../../../src/components/tableRow/TableRow';
import { H2, H3, P1, P2 } from '../../../src/components/texts/Texts';
import CenterLayout from '../../../src/layouts/centerLayout/CenterLayout';
import theme from '../../../src/theme/theme';

export default () =>
  storiesOf('Card List Item', module)
    .addDecorator((getStory) => (
      <CenterLayout color={theme.palette.primary}>
        {
          <View style={{ width: '100%', paddingLeft: 17, paddingRight: 17 }}>
            {getStory()}
          </View>
        }
      </CenterLayout>
    ))
    .add('Default', () => (
      <CardListItem>
        <H2>Default</H2>
      </CardListItem>
    ))
    .add('Three words', () => (
      <>
        <View
          style={{
            widht: '100%',
            paddingLeft: 13,
            paddingRight: 13,
            marginBottom: 10,
          }}
        >
          <TableRow
            columns={[
              { component: <P2>Card</P2>, align: 'left' },
              { component: <P2>Balance</P2>, align: 'right' },
              { component: <P2>Utilization</P2>, align: 'right' },
            ]}
          />
        </View>
        <CardListItem>
          <TableRow
            columns={[
              { component: <P1>Macy's (5859)</P1>, align: 'left' },
              { component: <H3>$6,500,34</H3>, align: 'center' },
              { component: <H3>10%</H3>, align: 'center' },
            ]}
          />
        </CardListItem>
      </>
    ));
