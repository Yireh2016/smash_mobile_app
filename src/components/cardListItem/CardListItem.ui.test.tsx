import * as React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import CardListItem from './CardListItem';

describe('CardListItem renders correctly', () => {
  it('CardListItem light version', () => {
    const ComponentProps = {
      children: <View />,
      isLight: false,
    };
    const Element = () => {
      return <CardListItem {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('CardListItem default version', () => {
    const ComponentProps = {
      children: <View />,
    };
    const Element = () => {
      return <CardListItem {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
