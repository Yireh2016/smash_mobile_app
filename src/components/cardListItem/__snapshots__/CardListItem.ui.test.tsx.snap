// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListItem renders correctly CardListItem default version 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#142836",
      "borderRadius": 12,
      "width": "100%",
    }
  }
>
  <View />
</View>
`;

exports[`CardListItem renders correctly CardListItem light version 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#142836",
      "borderRadius": 12,
      "width": "100%",
    }
  }
>
  <View />
</View>
`;
